scipy/optimize/_direct.py

```diff
@@ -0,0 +1,203 @@
+    ("The best function value found is within {} percent "
+     "of the (known) global optimum f_min"),
+    ("The volume of the hyper-rectangle with best function value found "
+     "is below vol_per={}"),
+    ("The measure of the hyper-rectangle with best function value found "
+     "is below sigma_per={}"),
+)
+
+
+def direct(
+    func: Callable[[npt.ArrayLike, Tuple[Any]], float],
+    bounds: Union[Iterable, Bounds],
+    *,
+    args: tuple = (),
+    disp: bool = False,
+    iatol: float = 1e-4,
+    maxfun: int = 20000,
+    maxiter: int = 6000,
+    locally_biased: bool = True,
+    f_min: float = -np.inf,
+    f_min_per: float = 0.01,
+    vol_per: float = -1.0,
+    sigma_per: float = -1.0,
+    callback: Optional[Callable[[npt.ArrayLike], NoneType]] = None
+) -> OptimizeResult:
+    r"""
+    Solve an optimization problem using the DIRECT
+    (Dividing Rectangles) algorithm.
+
+    The algorithm is due to Jones et al. [1]_.
+
+    Parameters
+    ----------
+    func: callable
+        The objective function to be minimized.
+        ``func(x, \*args) -> float``
+        where ``x`` is an 1-D array with shape (n,) and args is a tuple of
+        the fixed parameters needed to completely specify the function.
+    bounds : sequence or `Bounds`
+        Bounds for variables. There are two ways to specify the bounds:
+        1. Instance of `Bounds` class.
+        2. ``(min, max)`` pairs for each element in ``x``, defining the finite
+        lower and upper bounds for the optimizing argument of `func`. It is
+        required to have ``len(bounds) == len(x)``. ``len(bounds)`` is used
+        to determine the number of parameters in ``x``.
+    args : tuple, optional
+        Any additional fixed parameters needed to
+        completely specify the objective function.
+    disp : bool, optional
+        Prints the evaluated `func` at every iteration.
+    iatol : float, optional
+        Ensures sufficient decrease in function value when a new potentially
+        optimal hyperrectangle is chosen. Default is 0.0001.
+    maxfun : int, optional
+        Approximate upper bound on objective function evaluations.
+        Default is 2000.
+    maxiter : int, optional
+        Maximum number of iterations. Default is 6000.
+    locally_biased : bool, optional
+        Whether to use the original [1]_ or modified [2]_ DIRECT algorithm.
+        Default is True. Possible values:
+
+        * ``locally_biased=False`` - use the original DIRECT algorithm
+        * ``locally_biased=True`` - use the modified DIRECT-l algorithm
+    f_min : float, optional
+        Function value of the global optimum.
+        By default it is a negative value whose absolute value is very large.
+        Set this value only if the global optimum is known.
+    f_min_per : float, optional
+        Terminate the optimization when the percent error satisfies:
+
+        .. math::
+
+            100*(f_{min} - f_{global})/\max(1, |f_{global}|) \leq f_{glper}
```

Comment:
It seems that "f_{global}" corresponds with `f_min`, "f_{glper}" corresponds wth `f_min_per`, and "f_min" is the current best value? It would be better to define the symbols or at least make the correspondence more direct (e.g. "f_{min}" correspond with `f_min`). My preference would be to use single-letter subscripts (so that we don't need to explicitly un-italicize the words "global" and "min") and define the correspondence, e.g.

```suggestion
            (f_i - f^*)/\max(1, |f^*|) \leq \texttt \tt{f_min_per}/100
            
        where :math:`f_i` is the current value of the objective function and
        :math:`f^*` is the global optimum given by `f_min`.
```
(This assumes that others are OK with the use of LaTeX here. I'm fine with it, but use is discouraged outside of Notes - and even within Notes, depending on who you ask.)